src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -252,9 +253,8 @@ private[health] class HealthCheckActor(
     // dead task (in addition to the living one), thus sometimes leading
     // Marathon to report the task / instance / app as unhealthy while
     // everything is running correctly.
-    val healthOfInstanceId = healthByTaskId.find(_._1.instanceId == instanceId)
-    if (healthOfInstanceId.isDefined)
-      healthByTaskId.remove(healthOfInstanceId.get._1)
+    val healthToRemove = healthByTaskId.get(taskId)
```

Comment:
This re-introduces the bug fixed originally here: marathon keeps an instance id but tries several time to schedule the task on Mesos (so TaskId are cycling). If actors receives an health result from a previous task, under the same instanceId, it will be kept indefinitely...
The original patch was "purging" the previous TaskId health results by looking by instanceId, which is something we need to keep